### PR TITLE
A number of journal fixes

### PR DESCRIPTION
**Pull request by poettering**
Various fixes for the journal, in particular covering #4088 #4278 #4059 #4060

### CODE REVIEW

File: src/journal/journald-server.c [hunk 1]
```diff
@@ -595,52 +595,86 @@ static void server_cache_hostname(Server *s) {
...
         case -ESHUTDOWN:       /* Already archived              */
                 log_warning("%s: Journal file corrupted, rotating.", f->path);
                 return true;
+
         case -EIDRM:           /* Journal file has been deleted */
                 log_warning("%s: Journal file has been deleted, rotating.", f->path);
                 return true;
+
+        case -ETXTBSY:         /* Journal file is from the future */
+                log_warning("%s: Journal file is from the future, rotateing.", f->path);
```
Comment (AsciiWolf) [inline src/journal/journald-server.c:633]: > rotat**e**ing

Typo?

**Comment by hese10** [comment]
I tested these changes for that hanging "--list-boots" case (#4278), and these changes prevented the forever loop in journalctl.

**Comment by keszybz** [comment]
Looks all good. Typo fixed in merge.

**Comment by hese10** [comment]
It seems that these journald fixes do not help at least in openstack environment where we have different time zone (+8h into future) during boot up until VM gets correct time via NTP. In these cases we end up still hanging "journalctl --list-boots" command. Also we have problem with cursor parameter in journalctl in these cases.

**Comment by hese10** [comment]
It seems "--list-boots" hangs due to corrupted journal file ("bad message" error for verify).

**Comment by poettering** [comment]
@hese10 if you want your issues not being ignored, file proper issues for them. First reproduce them on the most recent systemd version, and then do provide a minimal set of journal files that exposes the misbheaviour you are seeing. Thanks. If you just comment on some already-closed PR nobody will take notice.

**Pull request merged**